### DISCUSSION TITLE
dynamic draw flow chart basic using joint JS

### DISCUSSION THREAD

**Discussion by adarshjalaja** [clientIO/joint#1865] [Q&A]
is it possible  to dynamically draw a flow chart using joint JS, attach custom attributes and export data as JSON
using JOINT JS alone. (not commercial JOINT JS +)

**Comment by kumilingus** [comment]
Yes, that's what *JointJS* (open source) is good at.

- Add new shapes: https://jsfiddle.net/x0tb26gn/
- Connect shapes with links: https://codepen.io/jointjs/pen/XWzdOQo
- Custom attrributes: https://resources.jointjs.com/tutorial/custom-attributes
- Export to JSON: https://resources.jointjs.com/tutorial/serialization

**Reply by adarshjalaja** [reply]
thanks. a lot.

I have one simple query in addn. is there a decision box element (yes/no decision) by default or we need to custom do circles and arrows or / custom link design ?

**Reply by kumilingus** [reply]
The decision shape example can be found here:
https://resources.jointjs.com/demos/diagram-index
https://resources.jointjs.com/demos/fta

It can be created from scratch or you can use built-in generic elemetns [`standard.Path` or `standard.Polyline`](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#shapes.standard) and draw the rhombus shape by using the `d` resp. `points` attribute.

```js
const polyline = new joint.shapes.standard.Polyline();
polyline.resize(100, 100);
polyline.attr('body/points', '0,0 0,calc(h) calc(w),calc(h) calc(w),0');
polyline.addTo(graph);
```

See the tutorial for more info (https://resources.jointjs.com/tutorial/elements, https://resources.jointjs.com/tutorial/special-attributes),

**Reply by adarshjalaja** [reply]
is diagram index a joint JS + library UTIL

**Reply by kumilingus** [reply]
Sorry, it is *JointJS+*. But [this is](https://github.com/clientIO/joint-plus-tutorial-react/blob/mui-tree-view/src/tree-data.ts#L32-L43) how the decision shape was created there.

**Reply by adarshjalaja** [reply]
is it possible to draw a multi direction LINK using join JS lib alone ? for eg like a YES NO branch from a element to 2 branches , is it custom doable using basic joint JS lib (open source) ?

**Reply by kumilingus** [reply]
In JointJS, a link has a single source and single target. If you want to create a "fork" link, you need to emulate it with two links.

**Reply by adarshjalaja** [reply]
thanks a lot . appreciate your help. !!!
